File: richpath/src/main/java/com/richpath/pathparser/PathParser.java
package com.richpath.pathparser;

import android.graphics.Path;

/**
 * Created by tarek on 6/27/17.
 */

public class PathParser {

    /**
     * @param pathData The string representing a path, the same as "d" string in svg file.
     * @return the generated Path object.
     */
    public static Path createPathFromPathData(String pathData) {

        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
            return PathParserCompatApi21.createPathFromPathData(pathData);
        } else {
            return PathParserCompat.createPathFromPathData(pathData);
        }
    }

    public static void createPathFromPathData(Path path, String pathData) {
        PathParserCompat.createPathFromPathData(path, pathData);
    }

}

File: richpath/src/main/kotlin/com/richpath/pathparser/PathParser.kt
package com.richpath.pathparser

import android.graphics.Path
import android.os.Build

object PathParser {

    /**
     * @param pathData The string representing a path, the same as "d" string in svg file.
     * @return the generated Path object.
     */
    fun createPathFromPathData(pathData: String?): Path {
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            PathParserCompatApi21.createPathFromPathData(pathData) ?: Path()
        } else {
            PathParserCompat.createPathFromPathData(pathData)
        }
    }

}